app/models/lecture.rb

```diff
@@ -1,3 +1,16 @@
 class Lecture < ApplicationRecord
   belongs_to :question_set
+  
+  #compute out of all the responses, how many corresponded to each answer
+  #returns hash keyed by answer id with value for % responses with that answer id
+  def self.compute_percentages (q)
+    @responses = Response.where(question: q)
+    @answers = Answer.where(question: q)
+    @total = @responses.length
+    hash = Hash.new(0)
```

Comment:
This is done specifically to ensure that there's an entry for answers with 0 responses. Otherwise the only answers in the hash table are ones with responses for them.